File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int r = 0;
        int l = nums.size()-1;
        binarySearch(nums, target, r, l);
    }

private:
    int binarySearch(vector<int>&nums, int target, int r, int l) {
        if(r>l){
            return -1;
        }
        int m = r + (l-r)/2;

        if (nums[m] == target) {
            return m;
        }
        if (nums[m] > target) {
            return binarySearch(nums, target, r, m - 1);
        }
        return binarySearch(nums, target, m + 1, l);
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        p = len(nums) - 1
        for i in range(len(nums) - 1):
            if nums[i] > nums[i+1]:
                p = i
                break 
        
        if p == len(nums) - 1:
            return nums[0]
            
        return nums[p + 1]